pandas/tests/groupby/test_any_all.py

```diff
@@ -61,6 +65,28 @@ def test_any():
     tm.assert_frame_equal(result, expected)
 
 
+def test_any_non_keyword_deprecation():
+    df = DataFrame({"A": [1, 2], "B": [0, 2], "C": [0, 0]})
+    msg = (
+        "In a future version of pandas all arguments of "
+        "DataFrame.any will be keyword-only"
+    )
+    with tm.assert_produces_warning(FutureWarning, match=msg):
```

Comment:
@MarcoGorelli Where to add the Future Warning in the function any. 
Currently, the tests are not passing.

Reply:
I'd say in pandas/core/generic.py

Reply:
I have made the changes. 
I think, the deprecation message is incorrect, it should mention `DataFrame.any` or `Series.any`, rather than `NDFrame.any`.

Reply:
ah, yes - then you'll want to add the warning to `frame.py` and `series.py` just before`super.any()` is called

Reply:
Moving `@deprecate_non_keyword` to `series.py` and `frame.py`, doesn't seem to trigger the Future warning.

What could be wrong here?

Reply:
Not sure off the top of my head - can you check what's done in other similar functions where this happens? else I'll have a look later

Reply:
.